src/main/java/picard/illumina/parser/NewIlluminaDataProvider.java

```diff
@@ -12,58 +11,38 @@
-    private final Map<String, BarcodeMetric> barcodeMetricMap;
-    private final int maxNoCalls;
-    private final int maxMismatches;
-    private final int minMismatchDelta;
-    private final int minimumBaseQuality;
-    private final BarcodeMetric noMatchMetric;
+    private BarcodeFileReader barcodeReader = null;
 
     /**
      * Construct a NewIlluminaDataProvider to create a ClusterData iterator over all clusters for a given tile.
      *
      * @param cbcls              A list of cbcls to use when creating this data provider.
      * @param filterFiles        A list of the pf filter files to use when creating this data provider.
-     * @param barcodesMetrics    A map of barcode to barcode metrics to output for this data provider.
-     * @param maxNoCalls         The maximum number of no calls to allow in a match.
-     * @param maxMismatches      The maximum number of mismatched calls to allow in a match.
-     * @param minMismatchDelta   The minimum mismatch difference between the best and second best matches.
-     * @param minimumBaseQuality The minimum base quality to allow for a matching base call.
      */
     NewIlluminaDataProvider(final List<File> cbcls, final List<AbstractIlluminaPositionFileReader.PositionInfo> locs,
-                            final File[] filterFiles, final int lane, final int tileNum, final OutputMapping outputMapping,
-                            boolean usingQualityScores, final Map<String, BarcodeMetric> barcodesMetrics,
-                            final BarcodeMetric noMatchMetric,
-                            int maxNoCalls, int maxMismatches, int minMismatchDelta, int minimumBaseQuality) {
+                            final File[] filterFiles, final int lane, final int tileNum,
+                            final OutputMapping outputMapping, final File barcodeFile) {
         super(lane, outputMapping);
-        this.usingQualityScores = usingQualityScores;
-        this.barcodeMetricMap = barcodesMetrics;
-        this.maxNoCalls = maxNoCalls;
-        this.maxMismatches = maxMismatches;
-        this.minMismatchDelta = minMismatchDelta;
-        this.minimumBaseQuality = minimumBaseQuality;
+
         Map<Integer, File> filterFileMap = new HashMap<>();
         for (File filterFile : filterFiles) {
             filterFileMap.put(fileToTile(filterFile.getName()), filterFile);
         }
-        this.reader = new CbclReader(cbcls, filterFileMap, outputMapping.getOutputReadLengths(), tileNum, locs, false);
-        this.outputReadStructure = outputMapping.getOutputReadStructure();
-
-        this.noMatchMetric = noMatchMetric;
+        this.reader = new CbclReader(cbcls, filterFileMap, outputMapping.getOutputReadLengths(), tileNum, locs, outputMapping.getOutputCycles(), false);
+        if (barcodeFile != null) {
+            this.barcodeReader = new BarcodeFileReader(barcodeFile);
+        }
     }
 
     @Override
     public void close() {
+        reader.clear();
```

Comment:
clear could be part of the close but we need to clear out the cache since the thread executor jobs hold reference to these readers.